feat: add skeleton of step function class and example stack (#290)
DataDog/datadog-cdk-constructs: examples/step-functions-typescript-stack/README.md

> # Datadog CDK TypeScript Example

Author: Written from scratch

> This example is used for testing the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library for Step Functions. The Step Functions support is still being built. Once it's ready, this example will be merged into the `typescript-stack` so that a single stack can be used for testing both Lambda functions and Step Functions, making it easier for CDK construct developers to test changes and ensure that their changes don't break things. The two examples are kept separate for now so that the incomplete Step Function support won't break the testing of Lambda functions.

Reviewer: ```suggestion
This example is used to test the [datadog-cdk-constructs](https://github.com/DataDog/datadog-cdk-constructs) v2 library for Step Functions. The Step Functions support is still in development. Once it's ready, this example will be merged into the `typescript-stack` so that a single stack can be used for testing both Lambda functions and Step Functions, making it easier for CDK construct developers to test their changes. The two examples are kept separate for now, so the incomplete Step Function support won't break the testing of Lambda functions.
```
I skipped the explanation of why someone would test, just because your audience likely already knows that. But if you wanted to keep it in, you can leave that part. I might say "easier for CDK construct developers to ensure that their changes don't cause regressions" or something along those lines, just for simplicity.

Author: Make sense! I prefer simplicity as well. Will remove the redundant part.
